src/hooks/useComplianceState.ts

```diff
@@ -0,0 +1,57 @@
+import { shallowEqual, useSelector } from 'react-redux';
+
+import { ComplianceStatus } from '@/constants/abacus';
+import { ComplianceStates } from '@/constants/compliance';
+import { BLOCKED_COUNTRIES, CountryCodes, OFAC_SANCTIONED_COUNTRIES } from '@/constants/geo';
+import { STRING_KEYS } from '@/constants/localization';
+import { isMainnet } from '@/constants/networks';
+
+import { getComplianceStatus, getGeo } from '@/state/accountSelectors';
+
+import { useStringGetter } from './useStringGetter';
+
+export const useComplianceState = () => {
+  const stringGetter = useStringGetter();
+  const complianceStatus = useSelector(getComplianceStatus, shallowEqual);
+  const geo = useSelector(getGeo, shallowEqual);
+
+  let complianceState = ComplianceStates.FULLACCESS;
+
+  let complianceMessage;
+
+  if (
+    complianceStatus === ComplianceStatus.FIRST_STRIKE ||
+    complianceStatus === ComplianceStatus.CLOSE_ONLY
+  ) {
+    complianceState = ComplianceStates.CLOSE_ONLY;
+  } else if (
+    complianceStatus === ComplianceStatus.BLOCKED ||
+    (geo &&
+      [...BLOCKED_COUNTRIES, ...OFAC_SANCTIONED_COUNTRIES].includes(geo as CountryCodes) &&
+      isMainnet)
+  ) {
+    complianceState = ComplianceStates.READ_ONLY;
+  }
+
+  if (complianceStatus === ComplianceStatus.FIRST_STRIKE) {
+    complianceMessage = `${stringGetter({ key: STRING_KEYS.CLICK_TO_VIEW })} →`;
```

Comment:
nit: I sort of prefer using switch statements when there's a bunch of cases like this (because it makes it more explicit that all edge cases are handled), but will defer to you if it's worth it here